File: FoosballGames.Contracts/GetFoosballGame.cs
using FoosballGames.Infrastructure.Messaging;

namespace FoosballGames.Contracts;

public record GetFoosballGame(Guid Id) : IQuery<FoosballGame>;

public record FoosballGame
(
    Guid Id,
    DateTime Start,
    DateTime? End,
    IReadOnlyCollection<Set> Sets,
    bool Finished,
    Team? WinnerTeam
);

public record Set(int Number, bool Finished, byte RedTeamScore, byte BlueTeamScore);

File: FoosballGames.Contracts/Old/GetFoosballGame.fs
namespace FoosballGames.OldContracts

open System
open System.Collections.Generic
open FoosballGames.Infrastructure.Messaging

type Set = {Number: int; Finished: bool; RedTeamScore: byte; BlueTeamScore: byte}
type FoosballGame =
    {Id: Guid
     Start: DateTime
     End: DateTime Nullable
     Sets: IReadOnlyCollection<Set>
     Finished: bool
     WinnerTeam: Team Nullable}
    
type GetFoosballGame = { Id: Guid } interface IQuery<FoosballGame>